src/components/common/buttons/ClaimButton.js

```diff
@@ -67,11 +73,30 @@ const ClaimButton = withStyles(getStylesFromProps)(({ screenProps, styles, style
 })
 
 const AnimatedClaimButton = ({ screenProps, styles, animated, animatedScale }) => {
-  const [pushButtonTranslate, setPushButtonTranslate] = React.useState({})
+  const containerRef = useRef()
+  const [pushButtonTranslate, setPushButtonTranslate] = useState({})
+
+  const measureView = useCallback(async view => {
+    const initialMeasurement = await measure(view)
+
+    if (!initialMeasurement.width && !initialMeasurement.height) {
```

Comment:
1. why if the second measurement will fail ? probably we have to retry once layout will be actually updated ?
2. why the check is !initialMeasurement.width && !initialMeasurement.height ? Why do not:

- get initial measurement on mount effect and store to ref
- on claim queue status changed, measure and compare with stored
- if measurement still matches await delay, repeat previous step
- update measurement stored in the ref

Reply:
@serdiukov-o-nordwhale please review again. as we talked I made the measure func recursive so it keeps trying but this time with a much lower delay